tools/compogen/pkg/gen/readme.go

```diff
@@ -255,12 +283,20 @@ func parseResourceProperties(o *objectSchema) []resourceProperty {
 	}
 
 	props := make([]resourceProperty, len(o.Properties))
-	for _, prop := range propMap {
-		// We can safely access the order pointer because it has been
-		// previously validated by the caller.
-		props[*prop.Order] = prop
+	idx := 0
+	for k := range propMap {
+		props[idx] = propMap[k]
+		idx += 1
```

Comment:
```suggestion
	for k := range propMap {
		props[k] = propMap[k]
```

Or, alternatively

```go
	for k, prop := range propMap {
		props[k] = prop
```

Reply:
`props` is an array with a size of `len(o.properties)`, and the `order` might not be in the range `0:len(o.properties)`. Therefore, I changed the implementation to use sorting.

Reply:
Yeah when I implemented this I didn't think `order` didn't need to be consecutive or in range, so 👍 to using sorting here. My suggestion here is that using `idx` is unnecessary. `k` already starts at 0 and is incremented in each loop iteration.

Reply:
argh 🤦 right, sorry for the noise